wdl2cwl/main.py

```diff
@@ -36,784 +21,788 @@
+
+    def get_expr_apply(self, wdl_apply_expr: WDL.Expr.Apply) -> str:  # type: ignore
+        """Translate WDL Apply Expressions."""
+        function_name = wdl_apply_expr.function_name
+        arguments = wdl_apply_expr.arguments
+        if not arguments:
+            raise Exception(f"The '{wdl_apply_expr}' expression has no arguments.")
+        treat_as_optional = wdl_apply_expr.type.optional
+        if function_name == "_add":
+            left_operand, right_operand = arguments
+            right_operand = self.get_expr(right_operand)  # type: ignore
+            left_operand_value = self.get_expr(left_operand)
+            if getattr(left_operand, "function_name", None) == "basename":
+                treat_as_optional = True
+                referer = wdl_apply_expr.parent.name  # type: ignore
+            return (
+                f"{left_operand_value} + {right_operand}"
+                if not treat_as_optional
+                else f"{self.get_input(referer)} === null ? {left_operand_value} + {right_operand} : {self.get_input(referer)}"
+            )
+        elif function_name == "basename":
+            if len(arguments) == 1:
+                only_operand = arguments[0]
+                is_file = isinstance(only_operand.type, WDL.Type.File)
+                only_operand = self.get_expr_name(only_operand.expr)  # type: ignore
+                return (
+                    f"{only_operand}.basename"
+                    if is_file
+                    else f"{only_operand}.split('/').reverse()[0]"
+                )
+            elif len(arguments) == 2:
+                operand, suffix = arguments
+                is_file = isinstance(operand.type, WDL.Type.File)
+                operand = self.get_expr_name(operand.expr)  # type: ignore
+                suffix_str = self.get_wdl_literal(suffix.literal)  # type: ignore
+                regex_str = re.escape(suffix_str)
+                return (
+                    f"{operand}.basename.replace(/{regex_str}$/, '') "
+                    if is_file
+                    else f"{operand}.split('/').reverse()[0].replace(/{regex_str}$/, '')"
+                )
+        elif function_name == "defined":
+            only_operand = arguments[0]
+            only_operand = self.get_expr_name(only_operand.expr)  # type: ignore
+            return only_operand  # type: ignore
+        elif function_name == "_interpolation_add":
+            arg_value, arg_name = arguments
+            if isinstance(arg_name, WDL.Expr.String) and isinstance(
+                arg_value, WDL.Expr.Apply
+            ):
+                arg_name = self.get_expr(arg_name)  # type: ignore
+                arg_value = self.get_expr_apply(arg_value)  # type: ignore
+                return self.get_pseudo_interpolation_add(arg_value, arg_name)  # type: ignore
+            elif isinstance(arg_value, WDL.Expr.Get):
+                arg_name, arg_value = arg_value, arg_name
```

Comment:
This line is never called; is it still needed?

Reply:
It is used. It was used for the file you edited to make it possible for tests. bcftools_stats.wdl line 75

Reply:
While `bcftools_stats` is being tested, it no longer triggers this line. Please investigate. I'd like to get the code coverage back up to at least where it was prior.